Automattic/vip-block-data-api
# Introduce an analytics sampling rate for VIP sites

## Description

This PR will introduce a sampling algorithm for the analytics that sent, when this plugin is used on a VIP site. The sampling algorithm is as follows:

- Sample every 10s, or
- Sample every 10m.

This will ensure that we catch burst usage, or regular usage throughout the day without losing the core purpose behind the analytics.

Fixes #45 

## Steps to Test

1. Spin up a local dev using `vip dev-env`
2. Adjust the `is_wpvip_site()` to give back true
3. Pop a breakpoint inside `is_it_sampling_time`.
4. Call `http://<slug>.vipdev.lndo.site/wp-json/vip-block-data-api/v1/posts/1/blocks`, while keeping an eye on the time
5. Every 10m or 10s it should trigger the breakpoint, otherwise it shouldn't be triggered.
